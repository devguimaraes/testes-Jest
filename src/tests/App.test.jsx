/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import {
  fireEvent,
  getByLabelText,
  getByRole,
  getByTestId,
  getByText,
  render,
  screen,
} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import App, { calcularNovoSaldo } from "../App";
import { act } from "react-dom/test-utils";

describe("Componente Principal", () => {
  describe("Quando abrir app do banco", () => {
    it("O nome do banco é exibido", async () => {
      await render(<App />);

      const nomeDoBanco = screen.getByText("ByteBank");

      expect(nomeDoBanco).toBeInTheDocument();
    });

    it("Saldo inicial sendo exibido", async () => {
      await render(<App />);

      const saldoEmTela = screen.getByText("Saldo:");
      expect(saldoEmTela).toBeInTheDocument();
    });

    it("O botão de realizar operacao é exibido", async () => {
      await act(() => render(<App />));

      const botao = screen.getByRole("button");
      expect(botao).toBeInTheDocument();
    });
  });

  describe("Quando realizar a transação", () => {
    it("Quando realizar um saque, o saldo precisa diminuir", async () => {
      await render(<App />);

      const valores = {
        transacao: "saque",
        valor: 50,
      };

      const novoSaldo = calcularNovoSaldo(valores, 150);

      expect(novoSaldo).toBe(100);
    });

    it("Quando realizar o saque, a transacao deve ser realizada", async () => {
      await render(<App />);

      const saldo = screen.getByText("R$ 1000");
      const transacao = screen.getByLabelText("Saque");
      const valor = screen.getByTestId("valor");
      const botao = screen.getByRole("button");

      expect(saldo.textContent).toBe("R$ 1000");
      fireEvent.click(transacao, { target: { value: "saque" } });
      fireEvent.change(valor, { target: { value: 10 } });
      fireEvent.click(botao);

      expect(saldo.textContent).toBe("R$ 990");
    });

    it("Quando realizar um deposito o saldo precisa aumentar", async () => {
      await render(<App />);

      const valores = {
        transacao: "deposito",
        valor: 100,
      };

      const novaSaldo = calcularNovoSaldo(valores, 100);
      expect(novaSaldo).toBe(200);
    });
  });

  describe("Teste de snpashot do APP", () => {
    it("Verifica se elementos renderizam de forma correta", async () => {
      const { container } = await render(<App />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
