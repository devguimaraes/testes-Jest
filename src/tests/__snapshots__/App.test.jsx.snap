// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Principal Teste de snpashot do APP Verifica se elementos renderizam de forma correta 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <h1>
      ByteBank
    </h1>
  </header>
  <div
    class="Conta-header"
  >
    <h2>
      Conta
    </h2>
    <p>
      Saldo: 
      <span
        class="Saldo-valor"
        data-testid="saldo-conta"
      >
        R$ 1000
      </span>
    </p>
    <form>
      <div>
        <label>
          Depósito
          <input
            data-testid="transacao"
            name="transacao"
            type="radio"
            value="deposito"
          />
        </label>
      </div>
      <div>
        <label>
          Saque
          <input
            data-testid="transacao"
            name="transacao"
            type="radio"
            value="saque"
          />
        </label>
      </div>
      <label>
        Valor:
      </label>
      <input
        data-testid="valor"
        name="valor"
        type="text"
        value="0"
      />
      <div>
        <button
          type="submit"
        >
          Realizar operação
        </button>
      </div>
    </form>
  </div>
  <div
    data-testid="transacoes"
  />
</div>
`;
